packages/eslint-plugin/src/rules/no-dynamic-delete.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  TSESTree,
+  AST_NODE_TYPES,
+  TSESLint,
+} from '@typescript-eslint/experimental-utils';
+import * as tsutils from 'tsutils';
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'no-dynamic-delete',
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description:
+        'Bans usage of the delete operator with computed key expressions',
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      dynamicDelete: 'Do not delete dynamically computed property keys.',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  defaultOptions: [],
+  create(context) {
+    function createFixer(
+      member: TSESTree.MemberExpression,
+    ): TSESLint.ReportFixFunction | undefined {
+      if (
+        member.property.type === AST_NODE_TYPES.Literal &&
+        typeof member.property.value === 'string'
+      ) {
+        const { value } = member.property;
+        return fixer =>
+          fixer.replaceTextRange(
+            [member.property.range[0] - 1, member.property.range[1] + 1],
```

Comment:
one thing I just thought about - doesn't this fixer fail if there are spaces?
`delete foo[ 'bar' ]`

You'll have to use `sourceCode.getTokenBefore/After` to get the position of the leading/trailing `[]`